### PR TITLE
erlang@18: import from homebrew/versions

### PR DESCRIPTION
**Pull request by elliotthilaire**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/erlang@18.rb [hunk 1]
```diff
@@ -0,0 +1,119 @@
+# Major releases of erlang should typically start out as separate formula in
```
Comment (ilovezfs) [inline Formula/erlang@18.rb]: Please remove this comment.

File: Formula/erlang@18.rb [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+# Homebrew-versions, and only be merged to master when things like couchdb and
+# elixir are compatible.
+class ErlangAT18 < Formula
+  desc "Programming language for highly scalable real-time systems"
+  homepage "https://www.erlang.org/"
+  head "https://github.com/erlang/otp.git"
+
+  stable do
```
Comment (ilovezfs) [inline Formula/erlang@18.rb]: Please remove the `stable do` block and place `url` and `sha256` above where `head` is currently.

File: Formula/erlang@18.rb [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    # Download tarball from GitHub; it is served faster than the official tarball.
+    url "https://github.com/erlang/otp/archive/OTP-18.3.tar.gz"
+    sha256 "8d5436faf37a1273c1b8529c4f02c28af0eccde31e52d474cb740b012d5da7e6"
+  end
+
+  bottle do
```
Comment (ilovezfs) [inline Formula/erlang@18.rb]: Please remove the bottle block.

File: Formula/erlang@18.rb [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    sha256 "6a5c2f8702940ca205e9445db3c6915540d81b0ee088af48ab215a3a34689e16" => :mavericks
+  end
+
+  option "without-hipe", "Disable building hipe; fails on various macOS systems"
+  option "with-native-libs", "Enable native library compilation"
+  option "with-dirty-schedulers", "Enable experimental dirty schedulers"
+  option "with-java", "Build jinterface application"
+  option "without-docs", "Do not install documentation"
+
+  deprecated_option "disable-hipe" => "without-hipe"
```
Comment (ilovezfs) [inline Formula/erlang@18.rb]: The corresponding versions formula you're importing doesn't have either of these deprecated options, so please remove them.

File: Formula/erlang@18.rb [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+    sha256 "8d5436faf37a1273c1b8529c4f02c28af0eccde31e52d474cb740b012d5da7e6"
+  end
+
+  bottle do
+    cellar :any
+    rebuild 2
+    sha256 "40c0acf9675257003e2f590d5e0d2500c56f348e8631f97e05b5f825ed4ed36c" => :el_capitan
+    sha256 "b982d71869783c3615cb22fa0fd5bc689e071ab3960bc4f1679fb11108be7ed9" => :yosemite
+    sha256 "6a5c2f8702940ca205e9445db3c6915540d81b0ee088af48ab215a3a34689e16" => :mavericks
+  end
```
Comment (ilovezfs) [inline Formula/erlang@18.rb]: Please add `keg_only :versioned_formula`

File: Formula/erlang@18.rb [hunk 6]
```diff
@@ -0,0 +1,119 @@
+# Major releases of erlang should typically start out as separate formula in
+# Homebrew-versions, and only be merged to master when things like couchdb and
+# elixir are compatible.
+class ErlangAT18 < Formula
+  desc "Programming language for highly scalable real-time systems"
+  homepage "https://www.erlang.org/"
+  head "https://github.com/erlang/otp.git"
```
Comment (ilovezfs) [inline Formula/erlang@18.rb]: Not sure we really need any head spec for erlang@18 but if we are going to have one it should be
```
  head "https://github.com/erlang/otp.git", :branch => "maint-18"
```

I suggest you just remove it, though.

**Review by ilovezfs — DISMISSED** [review]
Please also create a PR to remove erlang-r18 from the homebrew/versions tap, and add corresponding entries to `tap_migrations.json` and `formula_renames.json`, as well as creating an erlang@19 alias to point to the current erlang formula.

Also, you'll need to backport the clock_gettime fix from the erlang formula to fix the build on 10.11: https://github.com/Homebrew/homebrew-core/blob/a3be92067ff4d9688c2cb921e6a39964a0f16794/Formula/erlang.rb#L47-L53

**Comment by elliotthilaire** [comment]
Didn't realise there was an corresponding version in homebrew-versions.
I will re-do this.

(Initial PR was based on erlang.rb when it was v18 in homebrew-core)

**Comment by elliotthilaire** [comment]
Corresponding homebrew/versions PR:
https://github.com/Homebrew/homebrew-versions/pull/1538

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
@elliotthilaire Merged! Thank you for doing this.

**Pull request closed without merging**